src/GNNGraphs/query.jl

```diff
@@ -404,6 +404,16 @@ function has_multi_edges(g::GNNGraph)
     length(union(idxs)) < length(idxs)
 end
 
+"""
+    khop_adj(g::GNNGraph,k::Int)
+
+Return ``A^k`` where ``A`` is the adjacency matrix of the graph 'g'.
+
+"""
+function khop_adj(g::GNNGraph,k::Int)
```

Comment:
we can follow the signature of https://github.com/CarloLucibello/GraphNeuralNetworks.jl/blob/bb479fa633eba2659c7cde3d5924429a2b98bf75/src/GNNGraphs/query.jl#L134
and add `T, dir, weighted`.